### ISSUE TITLE
Support file as parameters in jacococli

### ISSUE THREAD

**Issue by mauricioaniche** [jacoco/jacoco#1623]
### Scenario

Currently, jacococli requires the list of .exec files to be passed as arguments to the jacococli command. However, in very large monorepos, the list of files might be too big, causing an "Arguments too long" error in Linux systems.

One idea would be to pass a file containing the list of .exec files instead of one-by-one in the command line. 

It turns out args4j, the library that jacococli uses to parse command-line arguments, already supports this. However, the current dependency is 10-years old. Bumping it to a more recent version (last available is 2.37, in comparison to current 2.0.28) would already make it work (tested locally).

### Current Behaviour

Jacococli crashes due to "Arguments too long".

### Wanted Behaviour

Being able to pass a file containing the .exec files instead of having to pass them one-by-one.

We can achieve that by simply bumping the version of args4j.

Suggestion: I would make this feature as opt-in, to keep compatibility and for security reasons. So maybe an environment variable `JACOCOCLI_SOMETHING` that, when, true, enables this behavior in the args4j parsing. (This feature is enabled by default in args4j, so implementation would have to be the other way around: disable it by default, and enable it in presence of the variable).

When enabled, users can do `java -jar jacococli.jar report @exec-files-list.txt ...`.

I'd gladly implement it if you think it's a good idea.

### Possible Workarounds

None that I can think of.

**Comment by marchof** [comment]
> However, the current dependency is 10-years old.

This is because the minimum Java runtime version for JaCoCo is Java 5. Due to the wide usage of JaCoCo in legacy projects we have currently no plans to rise our minimum Java runtime version.

**Comment by mauricioaniche** [comment]
@marchof Oh, good point. I understand. I'll close this issue, hoping that it will serve as documentation for others trying to do the same!

TL;DR: If you can afford to run on earlier versions of Java, simply bumping args4j to most recent and recompiling jacococli, will give you the ability of passing files as parameters!

**Comment by Godin** [comment]
@mauricioaniche first of all nice to see such a well described proposal of enhancement 👍 and thank you for your understanding of difficulties ♥️

And FYI as a workaround you can merge exec-files into one prior to generation of report from it - either using "merge" command or by simple concatenation 😉
